libr/core/agraph.c

```diff
@@ -4046,6 +4060,106 @@ static bool toggle_bb(RCore *core, ut64 addr) {
 	return false;
 }
 
+typedef struct r_canvas_location {
+	int x;
+	int y;
+} RPosition;
+
+static const char *strstr_xy(const char *p, const char *s, int *x, int *y) {
+	const char *d = strstr (p, s);
+	if (!d) {
+		return NULL;
+	}
+	const char *q;
+	int nl = 0;
+	int nc = *x;
+	for (q = p; q < d; q++) {
+		if (*q == '\n') {
+			nl++;
+			*x = 0;
+			nc = 0;
+		} else {
+			nc++;
+		}
+	}
+	*x += nc;
+	*y += nl;
+	return d;
+}
+
+static char *old_word = NULL;
+static RList *word_list = NULL;
+static int word_nth = 0;
+static int x_origin = 0;
+
+static void nextword(RCore *core, RConsCanvas *can, const char *word) {
+	RListIter *iter;
+	if (word_list && !strcmp (word, old_word)) {
+		RPosition *pos = r_list_get_n (word_list, word_nth);
+		if (pos) {
+			word_nth++;
+		} else {
+			word_nth = 0;
+			pos = r_list_get_n (word_list, word_nth);
+		}
+		if (pos) {
+			can->sx = pos->x;
+			can->sy = pos->y;
+		}
+		return;
+	} else {
+		r_list_free (word_list);
+		word_list = r_list_newf (free);
+	}
+	if (!*word) {
+		return;
+	}
+	char *s = r_core_cmd_strf (core, "agf@e:scr.color=0@e:scr.utf8=0");
```

Comment:
can you get this by doing `r_core_visual_graph (core, NULL, NULL, false);` and reading the cons buffer?

Reply:
the cons buffer is chopped to the screen size. thats why im doing this. but after implementing it i realized it could be implemented by iterating over all the RANodes to find out if the body strstr's and then taking the node size + strstr_xy inside it. but i have too many open prs and so many refactorings on the works. that i would prefer to merge this as a new feature when all the issues/comments get fixed and oen an issue for doing what its commented in the issue description

Reply:
oh nvm it works :D